contents/blog/mobile-session-replay.md

```diff
@@ -0,0 +1,141 @@
+
+What took so long? Although we had the structure to ingest and playback replays, recording them in mobile apps is much trickier than in web apps. This post goes over why and how we finally managed to overcome them.
+
+## What's so difficult about mobile session replay?
+
+Others have complained about [the lack of good mobile replay options](https://medium.com/goodones/15-years-later-there-is-still-no-good-session-replay-for-ios-f8d335999737). Why is that the case?
+
+### 1. Multiple platforms
+
+The industry's big secret about web session replay is that it largely relies on a single open source library to work: [rrweb](https://github.com/rrweb-io/rrweb). It includes tools for recording web interactions and state changes, structuring session data, and playback. 
+
+Unfortunately, rrweb for mobile doesn't exist. To build mobile session replay, we needed to do all the work ourselves, and when compared to the web, this is a lot. Instead of a single JavaScript SDK, mobile requires multiple (like iOS, Android, React Native, and Flutter). 
+
+There are even breaking differences within platforms. For example, Jetpack Compose uses a compositional model for UI, which is different from Android's traditional view-based model. This means you need to develop separate ways of doing replays when using it. iOS has a similar problem with SwiftUI versus UIKit.
+
+### 2. Performance
+
+Phones are much less powerful than desktops. Because of this, we need to be much more sensitive about performance. 
+
+If you've ever tried to record your phone screen, you would know its impact on performance. Apps take longer to load, animations become choppy, reactivity degrades, and your phone heats up.
+
+Anything that degrades your user experience is not an option for many developers.
+
+### 3. Privacy
+
+A big difference between web and mobile apps is the DOM. 
+
+In web apps, this provides a hierarchical tree structure that represents and uniquely identifies elements. The web also has standardized elements, like `<input type="password">`. Combining these makes it easier to identify, mask, and exclude sensitive elements on the web.
+
+Mobile doesn't have standardized structures or elements. Accessibility identifiers are also inconsistently implemented. This means identifying, masking, and excluding content is a lot trickier.
+
+### 4. Testing
+
+We're big fans of [dogfooding](/product-engineers/dogfooding) at PostHog. Often, we are our own best users, but we don't have a mobile app. 
+
+This meant that during development we relied on demo and open source apps. This risks creating something that doesn't work well for larger, production-quality apps.
+
+Developing a high-quality mobile replay product means relying more on our users and their feedback.
+
+## Solving mobile session replay's big problems
+
+Here's how we tackled all four of these:
+
+### 1. Multiple platforms
+
+We took a simple approach to solving this problem: do the work to develop for each platform.
```

Comment:
This is a bit dry. Can we make it wittier/more interesting?

"We took a simple approach to solving this problem: do the damn work for each platform."